test/endecode_test.c

```diff
@@ -1214,8 +1263,40 @@ int setup_tests(void)
         OSSL_PARAM_END
     };
 
+    OPTION_CHOICE o;
+
+    while ((o = opt_next()) != OPT_EOF) {
+        switch (o) {
+        case OPT_CONTEXT:
+            default_libctx = 0;
+            break;
+        case OPT_PROVIDER_NAME:
+            prov_name = opt_arg();
+            break;
+        case OPT_CONFIG_FILE:
+            config_file = opt_arg();
+            break;
+        case OPT_TEST_CASES:
+            break;
+        default:
+            return 0;
+        }
+    }
+
+    if (default_libctx) {
+        if (!test_get_libctx(&testctx, NULL, config_file, &deflprov, prov_name))
+            return 0;
+    } else {
+        if (!test_get_libctx(&testctx, &nullprov, config_file, &deflprov, prov_name))
+            return 0;
+    }
```

Comment:
The default library context is NULL. Does this change mean it's not tested any more?

Reply:
Yeah I think I previously misunderstood how test_get_libctx() was working. It would be useful for it to also cover the case of loading a provider into the default libctx. Maybe by passing a NULL as the **libctx argument?

Reply:
I have no idea what the mechanics should be like, but regarding "It would be useful for it to also cover the case of loading a provider into the default libctx", I'd rather label that as a *must*, especially considering that most application will probably not go around creating new library contexts, but rather just load whatever in the default one.

Reply:
Done